src/io.c

```diff
@@ -328,24 +320,16 @@ Int GetInputLineNumber(void)
 const Char * GetInputLineBuffer(void)
 {
     GAP_ASSERT(IO()->Input);
-    return IO()->Input->line;
+    // first byte of Input->line is reserved for the pushback buffer, so add 1
+    return IO()->Input->line + 1;
 }
 
-// Get current line position. In the case where we pushed back the last
-// character on the previous line we return the first character of the
-// current line, as we cannot retrieve the previous line.
 Int GetInputLinePosition(void)
 {
-    if (STATE(In) == &IO()->Pushback) {
-        // Subtract 2 as a value was pushed back
-        Int pos = IO()->RealIn - IO()->Input->line - 2;
-        if (pos < 0)
-            pos = 0;
-        return pos;
-    }
-    else {
-        return STATE(In) - IO()->Input->line - 1;
-    }
+    GAP_ASSERT(IO()->Input);
+    // first byte of Input->line is reserved for the pushback buffer, so add 1
+    // subtract 1 from STATE(In) because TODO/FIXME
+    return (STATE(In) - 1) - (IO()->Input->line + 1);
```

Comment:
I wanted to explain what's going on here, but then realized I don't understand it anymore 😂 I'll have to carefully step through the code, I guess... but if anybody can explain it right now, feel free to tell

Reply:
I'm not sure either :)

Reply:
I gave up trying to understand it -- I think it simply was nonsense! IMHO it was a bug / misfeature. So I now dropped the -1, and instead adjusted `SyntaxErrorOrWarning` to cope with the changed result. Indeed, this now also gives values that make sense. For example:
```
gap> local;
Syntax error: expression expected
local;
^^^^^
```
now gives `startPos=0` and `pos=5` in `SyntaxErrorOrWarning`, while before it was -1 and 4.